### DISCUSSION TITLE
0.4.21

### DISCUSSION THREAD

**Discussion by ObservedObserver** [Kanaries/graphic-walker#193] [Announcements]
## What's Changed
* Refactor: split dataSource from graphic walker by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/149
* fix: make eaiser to import old format data by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/185
* fix: make map center effective by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/188
* fix: facet issue by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/187
* feat: pivot sorting by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/191
* fix: computed field with filter by @islxyqwe in https://github.com/Kanaries/graphic-walker/pull/192

**Full Changelog**: https://github.com/Kanaries/graphic-walker/compare/0.4.20...0.4.21

<hr /><em>This discussion was created from the release <a href='https://github.com/Kanaries/graphic-walker/releases/tag/0.4.21'>0.4.21</a>.</em>

**Comment by islxyqwe** [comment]
Breaking changes:
 - visualConfig is split into visualConfig(changes will make recalculation) and visualLayout(changes will not make recalculation).
you can still pass only visualConfig that exported from old version, and pass no visual Layout; but export function in storeRef will output visualConfig and visualLayout.
 - Data manager is not a part of <GraphicWalker /> and the hideDataSourceConfig prop is removed, and you must provide a datasource to use GraphicWalker as a component. if you want to use Graphic walker as component with Data manager, use the new component named "FullGraphicWalker".
